### PR TITLE
[65CE02] Fix 16-bit branch PC-relative offset encoding

### PR DESCRIPTION
**Pull request by mlund**
## Summary
- Fix `getRelativeMOSPCCorrection()` to use -1 correction for 65CE02 16-bit branches (PC+2 base), keeping -2 for 65816 BRL (PC+3 base)
- The 65CE02 computes branch target as `opcode_addr + 2 + offset` for both 8-bit and 16-bit forms; the assembler incorrectly used `opcode_addr + 3` (the 65816 convention)
- Query `Feature65CE02` from the fragment's `MCSubtargetInfo` to distinguish architectures
- Add backward branch, boundary (127/128), and 45GS02 coverage to MC test

## References
Confirmed against three independent sources:
- MEGA65 FPGA core (`gs4510.vhdl`, `B16TakeBranch` microcode)
- Xemu emulator (`cpu65.c`, `_BRA16` function)
- [Pagetable.com 65CE02 reference](https://pagetable.com/c64ref/6502/cpu_65ce02.txt)

## Test plan
- [x] `llvm-lit llvm/test/MC/MOS/long-branches-65ce02.s` — updated encoding + edge cases
- [x] `llvm-lit llvm/test/MC/MOS/long-branches-65816.s` — no regression for 65816 BRL
- [x] Full MOS MC + CodeGen test suite (117/117 pass)
- [x] vector_logo.prg runs correctly in xmega65 emulator
- [x] **Update**: Using a [patched SDK simulator with 65CE02 opcodes](https://github.com/mlund/llvm-mos-sdk/tree/sim-65ce02), this PR passes the complete [llvm-test-suite](https://github.com/llvm-mos/llvm-test-suite) and all tests in [mega65-freezer](https://github.com/mlund/mega65-freezer) a ~10k line C23 port of the original cc65 code.

Co-authored with Claude AI and humanly supervised and reviewed by @mlund in accordance with LLVM guidelines.

### CODE REVIEW

File: llvm/lib/Target/MOS/MCTargetDesc/MOSAsmBackend.cpp [hunk 1]
```diff
@@ -83,10 +83,15 @@ bool MOSAsmBackend::fixupNeedsRelaxation(const MCFixup &Fixup,
   return true;
 }
 
-static int getRelativeMOSPCCorrection(const bool IsPCRel16) {
-  // MOS's PC relative addressing is off by one or two from the standard LLVM
-  // PC relative convention.
-  return IsPCRel16 ? -2 : -1;
+static int getRelativeMOSPCCorrection(const bool IsPCRel16,
```
Comment (mysterymath) [inline llvm/lib/Target/MOS/MCTargetDesc/MOSAsmBackend.cpp]: Intrinsically, both pieces of information are necessary to make this determination, so neither should be optional.

**Review by mysterymath — APPROVED** [review]

File: llvm/lib/Target/MOS/MCTargetDesc/MOSAsmBackend.cpp [hunk 2]
```diff
@@ -83,10 +83,15 @@ bool MOSAsmBackend::fixupNeedsRelaxation(const MCFixup &Fixup,
...
+                                      const bool Is65CE02 = false) {
+  // LLVM measures PC-relative fixups from the fixup location (opcode_addr + 1).
+  // 8-bit branches: offset from opcode_addr + 2 → correction -1.
+  // 65CE02 16-bit branches: also offset from opcode_addr + 2 → correction -1.
+  // 65816 BRL: offset from opcode_addr + 3 (past full instruction) → correction -2.
+  if (IsPCRel16 && !Is65CE02)
```
[the remark this answers is not in the document]
Comment (mlund) [reply llvm/lib/Target/MOS/MCTargetDesc/MOSAsmBackend.cpp]: Both fixed: restored to ternary with a single comment on the actual distinction between the two targets.

File: llvm/lib/Target/MOS/MCTargetDesc/MOSAsmBackend.cpp (hunk 1 above)
Comment (mlund) [reply llvm/lib/Target/MOS/MCTargetDesc/MOSAsmBackend.cpp:81]: Fixed. `Is65CE02` no longer defaults.

**Comment by wbniv** [comment]
FYI: [#591](https://github.com/llvm-mos/llvm-mos/pull/591) adds out-of-range diagnostics on the same `MOSAsmBackend::applyFixup` hunk this PR fixes — complementary, not competing: this PR corrects the per-CPU PC-relative correction *value*, #591 range-checks the corrected value and turns silent offset truncation into a source-located error. The two will conflict textually, so #591 is written to yield: whenever this lands, I'll rebase it on top and the diagnostics will then guard the corrected offsets here too.
